api/clients/v2/examples/example_relay_retrieval_test.go

```diff
@@ -0,0 +1,86 @@
+package examples
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/status-im/keycard-go/hexutils"
+)
+
+// This example demonstrates how to use the RelayPayloadRetriever to retrieve a payload from EigenDA, running on
+// holesky testnet
+func Example_relayPayloadRetrieval() {
+	// You must provide a private key that either has a testnet reservation, or you must configure on-demand payments
+	// by sending funds to the payment vault.
+	privateKey := ""
+
+	// Create a payload disperser and disperse a sample payload to EigenDA
+	// This will be the payload we will later retrieve
+	payloadDisperser, err := createPayloadDisperser(privateKey)
+	if err != nil {
+		panic(fmt.Sprintf("create payload disperser: %v", err))
+	}
+	defer payloadDisperser.Close()
+
+	payload, err := createRandomPayload(4 * 1024) // (4KB of random data)
+	if err != nil {
+		panic(fmt.Sprintf("create random payload: %v", err))
+	}
+
+	dispersalCtx, dispersalCancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer dispersalCancel()
+	eigenDACert, err := payloadDisperser.SendPayload(dispersalCtx, payload)
+	if err != nil {
+		panic(fmt.Sprintf("send payload: %v", err))
+	}
+
+	fmt.Printf("Successfully dispersed payload\n")
+
+	// Create a payload retriever to retrieve from EigenDA relays
+	payloadRetriever, err := createRelayPayloadRetriever()
+	if err != nil {
+		panic(fmt.Sprintf("create relay payload retriever: %v", err))
+	}
+	defer payloadRetriever.Close()
+
+	retrievalCtx, retrievalCancel := context.WithTimeout(context.Background(), 1*time.Minute)
```

Comment:
is 1 min actually needed/reasonable here? What is expected latency? maybe add a comment

Reply:
I tweaked all the timeouts in [313ed140](https://github.com/Layr-Labs/eigenda/pull/1454/commits/313ed140bed4634181fbc7e636cb0f2ee9ddf7c0), since you're right that I was overly cautious.

I hesitate to make claims on expected latency: performance of v2 is actively being improved, so I don't want to make a conservative claim that becomes outdated and sells us short. Better to set a timeout that's a bit overboard, and interested parties can evaluate contemporary performance later.

Reply:
Yeah that's fair